Financing budgeting/FIN-13, FIN-15

aninaslyan opened the pull request:
Implementation of FIN-13, FIN-15 tasks

aninaslyan committed:
FIN-13
Showing selected number of rows in the footer

aninaslyan committed:
FIN-6
Payables automatically calculated data. Date format in whole application. Adding LoadingSpinner component.

aninaslyan committed:
FIN-13
Adding fixed Total row in Payables table, making footer fixed. Adding generateArrayByRange util

aninaslyan committed:
FIN-6
Adding Payables fixed Total footer

aninaslyan committed:
FIN-13
Separating TooltipText and other components. Changing Loading styles according to the design

aninaslyan committed:
FIN-6
Making Payable control panel

aninaslyan committed:
FIN-13
Correcting Table style

aninaslyan committed:
FIN-15
Row Edit drawer

aninaslyan committed:
FIN-15
Row Edit drawer and components inside

aninaslyan committed:
FIN-15
Adding PaymentHistory, Notes accordions

aninaslyan committed:
FIN-15
style changes

aninaslyan committed:
FIN-15
Payment history states, Notes

aninaslyan committed:
FIN-15
Table warning fixes

aninaslyan committed:
FIN-15
Row edit style changes

igortokhunts reviewed (commented):
@aninaslyan jan i just whant to know, don't we have this icons in material ui?

igortokhunts commented on src/src_V2/components/loading-spinner/LoadingSpinner.module.css:
```diff
@@ -0,0 +1,78 @@
+    -ms-transform: rotate(360deg);
+    transform: rotate(360deg);
+  }
+}
```
@aninaslyan jan i saw the work you did, but the same question comes in my mind, don't we have this component in mui? I saw there are many loaders in mui.

aninaslyan replied:
@igortokhunts jan you are right, mui has many loading indicators. This is the loading design that our designers wanted to have in the whole application. They sent me [the link](https://codepen.io/WebSonata/pen/bRaONB).
But the design is not final. As they wanted to have a different design and in the future, they want to have a fixed overlay so nothing behind is clickable (they promised to create figma screen). That's why I agreed to use the custom design, not the material one.

igortokhunts replied:
If this loader we need to use in the entire app it's better to have it in the components folder, am i right? is it global for entire app?

aninaslyan replied:
It's already in components folder. Yes, it's for the entire app.

igortokhunts replied:
great

igortokhunts commented on src/src_V2/components/sidebar/Sidebar.module.css:
```diff
@@ -9,12 +9,16 @@
   height: 40px !important;
   top: 15px;
-  z-index: 1201;
+  z-index: 1;
```
@aninaslyan jan why we need this changes in siber?

aninaslyan replied:
z-index of Sidebar conflicted with the Drawer (for row edit) I created. Then I decided to change overlapping priorities (z-indexes) one by one with normal smaller numbers: Sidebar - 0, Sidebar's toggle button - 1. And other z-indexes can be added accordingly, not like z-index: 999999

igortokhunts replied:
got it

igortokhunts commented on src/src_V2/components/tooltip/TooltipText.jsx:
```diff
@@ -0,0 +1,19 @@
+      </div>
+    </Tooltip>
+  );
+}
```
note just for have a discussion for this component with you

igortokhunts replied:
Is this also global one?

aninaslyan replied:
Yes, it is also a global component. It is inside the correct directory right?

igortokhunts replied:
yeah it is, It seems good one but i just whant to know why it is a global?

aninaslyan replied:
It can be reused. This component works in cooperation with useHoverTooltip custom hook. So if the element doesn't fit its content, three dots and a tooltip will be shown. Feel free to customize it according to your use case, maybe it will be better to send the width of the element from the outside.

igortokhunts replied:
Got it, ok great.

igortokhunts commented on src/src_V2/modules/manager/financial/components/payables/Payables.jsx:
```diff
@@ -1,302 +1,75 @@
-    id: 'invoiceAttachment',
-    label: 'Invoice Attachment',
-  },
+  { id: 'paymentId', label: 'Payment ID' },
```
@aninaslyan jan i think we can create constant.js file for example in finanical folder level and put all constant related to finance there (even if the data is temporary and we could get it from backend in the future) how do you think?

igortokhunts commented on src/src_V2/modules/manager/financial/components/payables/Payables.module.css:
```diff
@@ -1,41 +0,0 @@
-  overflow: hidden;
-  text-overflow: ellipsis;
-  width: 100px;
-}
```
need we this empty file here?

aninaslyan replied:
When creating a component, I always create a css file near to it, to show it always has its css module and whenever a style is needed, it must be applied to that css file (not to another file), to keep encapsulation. This habit came from Angular. Just different approaches :D But will remove it.

igortokhunts replied:
got it, great thanks

igortokhunts commented on src/src_V2/modules/manager/financial/components/payables/control-panel/ControlPanel.jsx:
```diff
@@ -0,0 +1,27 @@
+import { Box, Button, InputBase, Paper } from '@mui/material';
+import FilterAltIcon from '@mui/icons-material/FilterAlt';
+import ViewColumnIcon from '@mui/icons-material/ViewColumn';
```
i think we can use named import here as we discussed.

aninaslyan replied:
@igortokhunts jan, unfortunately `@mui/icons-material` doesn't have named export of all these icons, it has only exports of default ones. More about this [here](https://mui.com/components/icons/). You can also look into our `Header.jsx` component of `NotificationsIcon` import.

igortokhunts replied:
@aninaslyan jan please try FilterAlt instead of FilterAltIcon and ViewColumn instead of ViewColumnIcon.

igortokhunts replied:
The`NotificationsIcon` is the only component here i think its ok this import style in this case but we can change it as well.

igortokhunts commented on src/src_V2/modules/manager/financial/components/payables/edit/drawer-content/notes/Notes.jsx:
```diff
@@ -0,0 +1,55 @@
+          value={row.notes}
+          fullWidth
+          InputProps={{
+            endAdornment: (
```
i have created Endornment component if you tink we can reuse it here you can made the coresponding changes and move that component to global src_V2/components folder hym?

igortokhunts commented on src/src_V2/modules/manager/financial/constants/payment-type.js:
```diff
@@ -0,0 +1,8 @@
+  2: 'Transfer',
+};
+
+export default PaymentType;
```
The constants sizes aren't so big i think we can put all this data in constants.js and remove files, how do you think?

igortokhunts commented on src/src_V2/styles/colors.css:
```diff
@@ -2,8 +2,17 @@
+  --success: #43a047; /* green[600] */
+  --error: #f44336; /* red[500] */
+  --warningDark: #e65100; /* orange[900] */
+  --overdue: #c2185b; /* pink[700] */
```
@aninaslyan jan i think this global styles folder or only the file in it can be located in app folder how do yo think?

aninaslyan replied:
@igortokhunts jan please review this part again, I've made changes. In case of any suggestions, pls tell me

igortokhunts commented on src/src_V2/components/tooltip/TooltipText.jsx:
```diff
@@ -0,0 +1,19 @@
+import { useRef } from 'react';
+import { Tooltip } from '@mui/material';
+
+import { useHoverTooltip } from '../../hooks';
```
Also i just noted tha hook folder in the global scope i think it is kinda util or somethihng like that and don't need to be in the app level folder(src/src_v2)

aninaslyan replied:
Is it okay to locate the hooks folder in `src_V2/modules/global/hooks/filename.js` as Tigran did?

aninaslyan committed:
FIN-15
Importing colors globally. Putting constants into separate files.

aninaslyan committed:
FIN-15
Material icon import correction. TooltipText maxWidth property

aninaslyan committed:
FIN-15
Row edit sidebar some changes

tatul-m committed:
FIN-13

DONE
Added Payment History Table

TODO
Resolve styles
Add Date Picker
Add File Upload
Remove Close on outside click
Show warning for unsaved changes on close
Add Server Side connection

tatul-m committed:
Merge branch 'FinancingBudgeting/FIN-13' of https://github.com/titancamp/travely-client into FinancingBudgeting/FIN-13

TigranYengiba04 commented on src/src_V2/components/loading-spinner/LoadingSpinner.jsx:
```diff
@@ -0,0 +1,9 @@
+import styles from './LoadingSpinner.module.css';
+
+export default function LoadingSpinner() {
```
I think we can use Mui's spinners if our designers do not impose on us to use a custom one.

aninaslyan replied:
They agreed, will change accordingly.

TigranYengiba04 commented on src/src_V2/components/tooltip/TooltipText.jsx:
```diff
@@ -0,0 +1,23 @@
+      <div
+        ref={textElementRef}
+        className={styles.tooltip}
+        style={{ maxWidth: maxWidth || '100px' }}
```
pls avoid using inline styles and use Box instead div

aninaslyan replied:
In this case, inline style can't be avoided. It's being applied dynamically (will make Box and sx={})

TigranYengiba04 replied:
yep, pls use mui styling solution there.

igortokhunts replied:
> In this case, inline style can't be avoided. It's being applied dynamically (will make Box and sx={})

@aninaslyan jan just for dinamic styling, you also can dinamicly add className and remove so with this approach we can avoid using inline ones.

TigranYengiba04 commented on src/src_V2/modules/manager/financial/components/payables/Payables.jsx:
```diff
@@ -1,302 +1,55 @@
-  );
-};
+  function processPayables(payables) {
+    const processedPayables = [...payables].map((payable) => {
```
why are you using rest operator there? It will not help because you are accessing to array's item object nested property. The rest operator doesn't make a deep copy.

TigranYengiba04 commented on src/src_V2/modules/manager/financial/components/payables/edit/drawer-content/payment-history/PaymentHistory.jsx:
```diff
@@ -0,0 +1,229 @@
+  return (
+    <TableCellWrapper>
+      {(() => {
+        switch (type) {
```
maybe we just need to write a mapper, like: 
```
const mapToTypes = {
  [inputTypes.text]: <TextField
                label={inputLabel}
                size="small"
                variant="outlined"
                value={input}
                onChange={(e) => setInput(e.target.value)}
              />
               ....
}
```

and avoid switch case?

tatul-m replied:
This code part will be changed. This was simple and fastest way to create some ui view

TigranYengiba04 commented on src/src_V2/modules/manager/financial/components/payables/edit/drawer-content/payment-history/PaymentHistory.jsx:
```diff
@@ -0,0 +1,229 @@
+                  labelId="demo-simple-select-label"
+                  id="demo-simple-select"
+                  value={rowValue}
+                  onChange={(e) => setInput(e.value)}
```
`                  onChange={({ value }) => setInput(value)}
`

TigranYengiba04 commented on src/src_V2/modules/manager/financial/components/payables/edit/drawer-content/payment-history/PaymentHistory.jsx:
```diff
@@ -0,0 +1,229 @@
+  const loading = false;
+  const [data, setData] = useState(history || []);
+
+  console.log(data);
```
pls remove

TigranYengiba04 commented on src/src_V2/modules/manager/financial/components/payables/edit/drawer-content/payment-history/PaymentHistory.jsx:
```diff
@@ -0,0 +1,229 @@
+        ...d,
+      ];
+    });
+  }, [data]);
```
what is the purpose of adding data as a dependency of useCallback?

TigranYengiba04 commented on src/src_V2/modules/manager/financial/components/payables/edit/drawer-content/payment-history/PaymentHistory.jsx:
```diff
@@ -0,0 +1,229 @@
+  }, [addPayment]);
+
+  return (
+    <Accordion className={commonStyles.accordion} expanded={true}>
```
just:
`    <Accordion className={commonStyles.accordion}>
`

TigranYengiba04 commented on src/src_V2/modules/manager/financial/components/payables/edit/drawer-content/payment-history/PaymentHistory.jsx:
```diff
@@ -0,0 +1,229 @@
+          />
+          <Typography className={commonStyles.detailsTxt}>
+            Payment History{' '}
+            {history?.length && (
```
if history.length will be 0 it will show 0 instead nothing, pls write like this:
`            {!!history?.length && (
`

tatul-m replied:
Maybe write history?.length > 0 ?

TigranYengiba04 commented on src/src_V2/modules/manager/financial/components/payables/edit/drawer-content/tour-details/TourDetails.jsx:
```diff
@@ -0,0 +1,52 @@
+          <Box className={styles.accordionDetailItems}>
+            <Box>Tour Status</Box>
+            <Box className={styles.accordionDetailTxt}>
+              {PaymentStatus[row.status].toUpperCase()}
```
pls, use css to make text upper case.

aninaslyan committed:
FIN-15
Table search

aninaslyan committed:
FIN-15
Changing common css file path

aninaslyan committed:
Merge branch 'dev' of https://github.com/titancamp/travely-client into FinancingBudgeting/FIN-13

# Conflicts:
#	src/src_V2/components/index.js
#	src/src_V2/components/sidebar/Sidebar.jsx
#	src/src_V2/modules/manager/supplier-management/components/noData/NoData.jsx

aninaslyan committed:
FIN-15
Search and row edit form. Adding package for using datepicker

aninaslyan committed:
FIN-15
Row edit form validation

tatul-m committed:
FIN-13
Temp commit for pull

tatul-m committed:
Merge branch 'FinancingBudgeting/FIN-13' of https://github.com/titancamp/travely-client into FinancingBudgeting/FIN-13

aninaslyan committed:
FIN-15
Notes clear text

aninaslyan committed:
Merge branch 'dev' of https://github.com/titancamp/travely-client into FinancingBudgeting/FIN-13

# Conflicts:
#	src/src_V2/components/sidebar/Sidebar.module.css
#	src/src_V2/modules/manager/supplier-management/components/partnership/style.module.css

aninaslyan committed:
Merge branch 'dev' of https://github.com/titancamp/travely-client into FinancingBudgeting/FIN-13

# Conflicts:
#	src/src_V2/components/index.js
#	src/src_V2/utils/helpers.js

tatul-m committed:
Merge branch 'FinancingBudgeting/FIN-13' of https://github.com/titancamp/travely-client into FinancingBudgeting/FIN-13

# Conflicts:
#	package-lock.json
#	src/src_V2/modules/manager/financial/components/payables/edit/drawer-content/payment-history/PaymentHistory.jsx

Resolved error when selecting DueDate
Added datepicker in PaymentHistory

tatul-m committed:
Merge branch 'FinancingBudgeting/FIN-13' of https://github.com/titancamp/travely-client into FinancingBudgeting/FIN-13

aninaslyan committed:
FIN-13
Row edit - Depending fields

aninaslyan committed:
FIN-13
Making Confirm Dialog reusable component. Payables Unsaved and Apply popups

aninaslyan committed:
FIN-13
Setting date engine at the root level. Confirm Dialog enhancement

aninaslyan committed:
FIN-13
Optimization. Style changes

aninaslyan committed:
FIN-15
Payment History

aninaslyan committed:
FIN-13
Payment History methods

aninaslyan committed:
FIN-13
Confirmation Delete popup. Some corrections

igortokhunts commented on src/src_V2/components/sidebar/Sidebar.module.css:
```diff
@@ -1,41 +1,33 @@
   width: 40px !important;
   height: 40px !important;
-  z-index: 1202;
+  z-index: 1;
```
@aninaslyan jan why we need to set z-index to 1?

igortokhunts commented on src/src_V2/hooks/use-table-data-generator.js:
```diff
@@ -1,7 +1,7 @@
 
 const isColumnVisible = (visibleColumns, key) =>
-    Object.prototype.hasOwnProperty.call(visibleColumns, key) && visibleColumns[key];
+  Object.prototype.hasOwnProperty.call(visibleColumns, key) && visibleColumns[key];
```
@aninaslyan jan coudn't we just write 
`visibleColumns[key]` ?

aninaslyan committed:
FIN-13
Moving hooks folder to utils

aninaslyan committed:
FIN-13
Changing icons to material ones

aninaslyan committed:
Merge branch 'dev' of https://github.com/titancamp/travely-client into FinancingBudgeting/FIN-13

# Conflicts:
#	src/src_V2/utils/constants.js

aninaslyan committed:
FIN-13
Resolving logo and hooks path

aninaslyan committed:
Merge branch 'dev' of https://github.com/titancamp/travely-client into FinancingBudgeting/FIN-13

# Conflicts:
#	src/src_V2/modules/manager/container/header/Header.jsx
#	src/src_V2/modules/manager/container/sidebar/Sidebar.module.css
#	src/src_V2/modules/manager/financial/components/payables/Payables.jsx
#	src/src_V2/modules/manager/reporting/index.jsx
